core/src/test/scripts/unit_tests_validate_data.py

```diff
@@ -1415,6 +1415,106 @@ def test_duplicate_line(self):
+
+        # Expected ERROR message due to missing Exon column
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(3, record.line_number)
+        self.assertEqual(4, record.column_number)
+        self.assertIn("No Ensembl transcript ID found.", record.message)
+
+        # Expected ERROR message due to missing Ensembl transcript column
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(6, record.line_number)
+        self.assertEqual(5, record.column_number)
+        self.assertIn("No exon found.", record.message)
+
+        # Expected ERROR message due to missing Exon column
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(8, record.line_number)
+        self.assertEqual(12, record.column_number)
+        self.assertIn("No exon found.", record.message)
+
+    def test_transcript_not_in_genome_nexus(self):
+        """Test whether the transcripts are validated correctly by checking Genome Nexus"""
+        self.logger.setLevel(logging.ERROR)
+        record_list = self.validate('data_structural_variants_transcript_not_in_genome_nexus.txt',
+                                    validateData.StructuralVariantValidator)
+
+        self.assertEqual(1, len(record_list))
+        record_iterator = iter(record_list)
+
+        # Expected ERROR message due to value "1500" in Site1_Exon in line 2
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual("TEST_TRANSCRIPT", record.cause)
+        self.assertIn("Ensembl transcript not found in Genome Nexus.", record.getMessage())
+
+    def test_exon_not_in_transcript(self):
+        """Test whether the exons are found in the transcript"""
+        self.logger.setLevel(logging.ERROR)
+        record_list = self.validate('data_structural_variants_exon_not_in_transcript.txt',
+                                    validateData.StructuralVariantValidator)
+
+        self.assertEqual(len(record_list), 2)
+        record_iterator = iter(record_list)
+
+        # Expected ERROR message due to value "1500" in Site1_Exon in line 2
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(2, record.line_number)
+        self.assertEqual("1500 not in ENST00000242365", record.cause)
+        self.assertIn("Exon is not found in rank of transcript", record.getMessage())
+
+        # Expected ERROR message due to value "2000" in Site2_Exon in line 4
+        record = record_iterator.next()
+        self.assertEqual(logging.ERROR, record.levelno)
+        self.assertEqual(4, record.line_number)
+        self.assertEqual("2000 not in ENST00000389048", record.cause)
+        self.assertIn("Exon is not found in rank of transcript", record.getMessage())
+
```

Comment:
Should you also check that the different type of data uploaded in the database is correct? E.g. `Site1_Exon` is an integer and not a string.